test/parallel/test-runner-mock-timers.js

```diff
@@ -232,6 +232,82 @@ describe('Mock Timers Test Suite', () => {
       });
     });
 
+    describe('setImmediate Suite', () => {
+      it('should keep setImmediate working if timers are disabled', (t, done) => {
+        const now = Date.now();
+        const timeout = 2;
+        const expected = () => now - timeout;
+        global.setImmediate(common.mustCall(() => {
+          assert.strictEqual(now - timeout, expected());
+          done();
+        }));
+      });
+
+      it('should work with the same params as the original setImmediate', (t) => {
+        t.mock.timers.enable(['setImmediate']);
+        const fn = t.mock.fn();
+        const args = ['a', 'b', 'c'];
+        global.setImmediate(fn, ...args);
+        t.mock.timers.tick(9999);
+
+        assert.strictEqual(fn.mock.callCount(), 1);
+        assert.deepStrictEqual(fn.mock.calls[0].arguments, args);
+      });
+
+      it('should not advance in time if clearImmediate was invoked', (t) => {
+        t.mock.timers.enable(['setImmediate']);
+
+        const fn = mock.fn();
+        const id = global.setImmediate(fn);
+        global.clearImmediate(id);
+        t.mock.timers.tick(200);
+
+        assert.strictEqual(fn.mock.callCount(), 0);
+      });
+
+      it('should advance in time and trigger timers when calling the .tick function', (t) => {
+        t.mock.timers.enable(['setImmediate']);
+
+        const fn = mock.fn();
```

Comment:
same here with `common.mustCall(1)`, ame goes for the rest of them 🙂

Reply:
added! Thanks Chemi!